### DISCUSSION TITLE
Feature request catch expression in future .Net release

### DISCUSSION THREAD

**Discussion by yaddly** [dotnet/csharplang#8450] [Language Ideas]
The .Net framework introduced a `switch` expression to replace the switch structure:
```
string result = domain switch
{
    "us" => "United States",
    "de" => "Germany",
    "sk" => "Slovakia",
    "hu" => "Hungary",

    _ => "Unknown"
};
```

This new expression got me thinking if it would not be ideal to have a `catch` expression as well.

```
try
{}
catch (error) => {
  error is FormatException => {},
  error is ArgumentNullException => {},
  error is IndexOutOfRangeException => {},
  _ => {}
}
```
The above feature request is more elegant than the conventional use of multiple `catch` statements.

**Comment by 333fred** [comment]
Similar to (but quite the same) as https://github.com/dotnet/csharplang/discussions/989.
